front/components/data_source/DocumentOrTableUploadOrEditModal.tsx

```diff
@@ -569,15 +652,15 @@ const TableUploadOrEditModal = ({
       return;
     }
 
-    setUploading(true);
+    setIsUpserting(true);
     try {
```

Comment:
Why do we need a try/catch here? Looks like nothing can throw, right?

Reply:
Actually I didn't work on the code for uploading tables, but I agree

Reply:
Don't be shy when you see non-sense, you are more than welcome to fixe them 🙃.

Reply:
Gotcha, you mentioned that this is a ~legacy codebase  so there are multiple parts that could benefit from a bit of cleaning, but I'm trying to find the right balance to not tackle everything at once.

Reply:
💯

Reply:
Jumping in to advertise this: https://www.notion.so/dust-tt/Engineering-e0f834b5be5a43569baaf76e9c41adf2?pvs=4#5d92c748016b488686c1577e86637c81

We enforce the use of Result<> over try/catch throughout the codebase :+1: 

This is legacy here of course but good to keep in mind!